src/gg.regression.unity.bots/Editor/Scripts/CodeGenerators/RGCodeGenerator.cs

```diff
@@ -107,818 +114,872 @@ public static void GenerateRGScripts()
-
-                // add global click button action
-                actionInfos.Add(new RGActionInfo()
-                {
-                    ActionClassName = typeof(RGAction_ClickButton).FullName,
-                    ActionName = "ClickButton",
-                    Parameters = new List<RGParameterInfo>()
-                });
+            do
+            {
+                var asm = stack.Pop();
 
-                /* TODO (REG-1476): Solve how to add hand written actions automatically
-                   this will help us avoid weird assembly references also.
+                yield return asm;
 
-                // add key press action
-                actionInfos.Add(new RGActionInfo()
-                {
-                    ActionClassName = typeof(RGAction_KeyPress).FullName,
-                    ActionName = "KeyPress",
-                    Parameters = new List<RGParameterInfo>()
+                foreach (var reference in asm.GetReferencedAssemblies())
+                    if (!list.Contains(reference.FullName))
                     {
-                        new ()
+                        try
                         {
-                            Name = "keyId",
-                            Type = "string",
-                            Nullable = false
-                        },
-                        new ()
+                            stack.Push(Assembly.Load(reference));
+                            list.Add(reference.FullName);
+                        }
+                        catch (Exception)
                         {
-                            Name = "holdTime",
-                            Type = "double",
-                            Nullable = true
+                            // somme assemblies can't be loaded (Like mac os code signing)
                         }
                     }
-                });*/
-
-                // if these have been associated to gameObjects with RGEntities, fill in their objectTypes
-                EditorUtility.DisplayProgressBar("Extracting Regression Games Agent Builder Data", "Populating Object types", 0.6f);
-                var stateAndActionJsonStructure = CreateStateAndActionJsonWithObjectTypes(statesInfos, actionInfos);
 
-                // update/write the json
-                EditorUtility.DisplayProgressBar("Extracting Regression Games Agent Builder Data", "Writing JSON files", 0.8f);
-                WriteJsonFiles(stateAndActionJsonStructure.Item1.ToList(), stateAndActionJsonStructure.Item2.ToList());
-
-                // create 'RegressionGames.zip' in project folder
-                EditorUtility.DisplayProgressBar("Extracting Regression Games Agent Builder Data", "Creating .zip file", 0.9f);
-                CreateJsonZip();
-            }
-            finally
-            {
-                EditorUtility.ClearProgressBar();
             }
+            while (stack.Count > 0);
```

Comment:
This may be my GitHub formatting things, but I don't see the body for this?

Reply:
this is a do while, you're looking at the end of the loop